airbnb_project/policies/policies/valid_registration_number_policy.py

```diff
@@ -0,0 +1,42 @@
+from ..models.policy_model import Policy
+import re
+
+class ValidRegistrationNumberPolicy(Policy):
+  """
+  Policy to check if the registration number is valid based on a specific pattern and year range.
+  """
+  
+  START_YEAR = 13
+  END_YEAR = 24
+  VALID_REGISTRATION_PATTERN = re.compile(r"^[0-9]{2}-[0-9]{6}$")
+  
+  def is_valid_registration_number(self, registration_number: str) -> bool:
+    """
+    Validate the registration number based on pattern and year range.
+
+    Args:
+        registration_number (str): The registration number to validate.
+
+    Returns:
+        bool: True if the registration number is valid, False otherwise.
+    """
+    if not registration_number:
+      return False
+
+    if self.VALID_REGISTRATION_PATTERN.match(registration_number):
+      registration_year = int(registration_number[0:2])
+      return self.START_YEAR <= registration_year <= self.END_YEAR
+    else:
+      return False
+    
+  def evaluate(self, registration_number: str) -> bool:
```

Comment:
just curious. Maybe I don't understand the context here but I believe we wouldn't need an extra function that just returns `is_valid_registration_number()`?

Reply:
I made the `evaluate` method required for all policy instances, so in this case it would just return `is_valid_registration_number()`, but it might be different for other policies. What do you think?

Reply:
Ok I see what's happening. In this case, I'd recommend we use **inheritance** and move **evaluate** to **Policy**. That way, all child classes can use or override that function. Here's an example:


```
class Policy():
    def evaluate(self, registration_number: str) -> bool:
        pass
```

Does that make sense?

Reply:
@gmartinsnull  I do have `evaluate` in Policy model in `policy_model.py` here:
```python
def evaluate(self) -> bool:
    """
    Abstract method to evaluate the policy
    """
    raise NotImplementedError("Policies must implement this method")
```
If evaluate is not implemented in inherited models, it throws error above.